pkg/controller/master/backup/restore.go

```diff
@@ -577,7 +589,7 @@ func (h *RestoreHandler) deleteOldPVC(vmRestore *harvesterv1.VirtualMachineResto
 
 func (h *RestoreHandler) restartVM(vm *kv1.VirtualMachine) error {
 	logrus.Infof("restarting the vm %s, current state running:%v", vm.Name, *vm.Spec.Running)
-	if vm.Spec.Running == nil || *vm.Spec.Running {
+	if vm.Spec.Running == nil || !*vm.Spec.Running {
```

Comment:
For restoring to existent VM, we need existent VM to be stopped first, so we need to restart it.

Reply:
I don't get it. Why do we restart a VM when we want it to be stopped?

Reply:
Correct me if I am wrong. I think users may want their VM to go back to running state if they restore to an existent VM.

Reply:
That makes sense, but we should call `start` instead of `restart`.

Reply:
Updated it. Thank you.

Reply:
On the second thought, I think the reason for restarting VM is to recreate the VM instance so that the new VM spec is applied. Is it expected to start a stopped VM instead of restarting a running VM here?

Reply:
From webhook, I think we can't restore to a running VM. We have to stop it first.

https://github.com/harvester/harvester/blob/b7e562d382019d15d5e4a0394b1dfb5afa6fe630/pkg/webhook/resources/restore/validator.go#L92-L94